lib/m3u8-parser.js

```diff
@@ -34,14 +37,33 @@ module.exports = class m3u8parser extends Writable {
   _write(chunk, encoding, callback) {
     var lines = chunk.toString('utf8').split('\n');
     if (this._lastLine) { lines[0] = this._lastLine + lines[0]; }
-    lines.forEach((line, i) => {
-      if (i < lines.length - 1) {
-        this._parseLine(line);
-      } else {
-        // Save the last line in case it has been broken up.
-        this._lastLine = line;
+    if (!this.options.youtubeFormatting) {
+      lines.forEach((line, i) => {
+        if (i < lines.length - 1) {
+          this._parseLine(line);
+        } else {
+          // Save the last line in case it has been broken up.
+          this._lastLine = line;
+        }
+      });
+    } else {
+      this._lastLine = lines[lines.length - 2];
+    }
+    if (this.first) {
+      this._firstLines = lines;
+      this.first = false;
+    }
+    callback();
+  }
+
+  _final(callback) {
+    if (this.options.youtubeFormatting) {
+      const firstIndex = this._firstLines.findIndex(a => a.includes('#EXTINF'));
+      for (let i = 0; i < firstIndex.length - 1; i++) {
```

Comment:
Loop through everything before the first audio mark so we get all of those juicy details about the stream